### PR TITLE
Hell and minor fixes

### PR DESCRIPTION
**Pull request by NottyNoz**
New hell themed theater
Minor change to calculation of player distance to the infinity gauntlet
Added baked beans worldmodel support for ponies

### CODE REVIEW

File: lua/autorun/client/hell.lua [hunk 1]
```diff
@@ -0,0 +1,234 @@
+if game.GetMap() == "cinema_swamp_v2" then
+
+	if CLIENT then
```
Comment (cannon) [inline lua/autorun/client/hell.lua]: don't need to check this, it's in client realm

File: lua/autorun/client/hell.lua [hunk 2]
```diff
@@ -0,0 +1,234 @@
+if game.GetMap() == "cinema_swamp_v2" then
```
Comment (cannon) [inline lua/autorun/client/hell.lua]: it's assumed, no need to check

File: lua/autorun/client/hell.lua [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+			base:SetSize(ScrW()*0.3,ScrH()*0.3)
+			base:SetPos(ScrW()*0.25,ScrH()*0.25)
+			base:SetTitle("Warning")
+			base:Center()
+			base:MakePopup()
+			
+			local basec = {}
+			local text = {}
+			
+			basec[1] = vgui.Create("Panel",base)
```
Comment (cannon) [inline lua/autorun/client/hell.lua]: These type of thing are easier to do by overriding panel.Paint(self, w, h)

File: lua/autorun/client/hell.lua [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+			corner1b = Vector(-.5,4096.5,Zup)
+			corner2a = Vector(-1022,4096.5,Zup)
+			corner2b = Vector(-1022,4096.5,Zdn)
+			corner3a = Vector(-.5,5118,Zdn)
+			corner3b = Vector(-.5,5118,Zup)
+			corner4a = Vector(-1022,5118,Zdn)
+			corner4b = Vector(-1022,5118,Zup)
+		end
+		
+		function MakeSprite(plypos,vec,width,minheight,maxheight)
```
Comment (cannon) [inline lua/autorun/client/hell.lua]: Name too generic, these functions are in the global namespace. Same with ChangeZ and PromptPlayer

File: lua/autorun/client/hell.lua [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+			local rad = math.atan2(plypos.y - vec.y,plypos.x - vec.x)
+			local y1 = vec.y - (width * math.cos(rad))
+			local y2 = vec.y + (width * math.cos(rad))
+			local x1 = vec.x + (width * math.sin(rad))
+			local x2 = vec.x - (width * math.sin(rad))
+			render.DrawQuad(Vector(x1,y1,minheight),Vector(x1,y1,maxheight),Vector(x2,y2,maxheight),Vector(x2,y2,minheight))
+		end
+		
+		hook.Add("PostDrawOpaqueRenderables","Hell",function()
```
Comment (cannon) [inline lua/autorun/client/hell.lua]: skip if drawing depth or drawing skybox (see PostDrawOpaqueRenderables params)

File: lua/autorun/server/hell.lua [hunk 6]
```diff
@@ -0,0 +1,14 @@
+if game.GetMap() == "cinema_swamp_v2" then
+
+	if SERVER then
+	
+		util.AddNetworkString("HellTeleport")
+		
+		net.Receive("HellTeleport",function(len,ply)
```
[the remark this answers is not in the document]
Comment (NottyNoz) [reply lua/autorun/server/hell.lua]: I wouldn't know where to start with a proper check other than a cooldown

[the remark this answers is not in the document]
Comment (NottyNoz) [reply lua/autorun/server/hell.lua]: nevermind I realized I can check if they are near the portal

**Comment by cannon** [comment]
trying to merge it properly

**Pull request closed without merging**